Sources/eppo/EppoClient.swift

```diff
@@ -215,7 +215,21 @@ public class EppoClient {
                 subjectAttributes: subjectAttributes,
                 expectedVariationType: UFC_VariationType.integer
             )
-            return Int(try assignment?.variation?.value.getDoubleValue() ?? Double(defaultValue))
+            
+            // If we got an assignment error, return the default value
+            if assignment?.flagEvaluationCode == .assignmentError {
+                return defaultValue
+            }
+            
+            // Get the double value and check if it's an integer
+            if let doubleValue = try? assignment?.variation?.value.getDoubleValue() {
+                if !doubleValue.isInteger {
+                    return defaultValue
+                }
+                return Int(doubleValue)
+            }
+            
+            return defaultValue
```

Comment:
I was not expecting to touch `getIntegerAssignment` but this matches up with how we handle non-integer doubles in `getIntegerAssignmentDetails`, which was written to pass all the test cases in our test data (see `EppoClientUFCTests.swift`), and uses the `.flagEvaluationCode` attribute that I added to the `getInternalAssignment` output